### ISSUE TITLE
pkg_version_info comparison operator error 

### ISSUE THREAD

**Issue by KryeKuzhinieri** [r-lib/rlang#1469]
Hi,

We have an R package whose description file is as following:

```
....
....
highcharter (== 0.8.2),
htmltools,
htmlwidgets (== 1.5.3),
httr (== 1.4.2),
...
...
```

Up to a couple of months ago, we did not have any issue while installing the package. However, in the last month or so, probably after we updated the R version, we keep getting the following error while documenting the package. 

```
Error in `check_installed()`:
! `compare` must be one of ">", ">=", "<", or "<=".
Backtrace:
    ▆
 1. └─devtools::document()
 2.   └─roxygen2::roxygenise(pkg$path, roclets)
 3.     └─roxygen2 (local) load_code(base_path)
 4.       └─pkgload::load_all(path, helpers = FALSE, attach_testthat = FALSE)
 5.         └─pkgload:::load_depends(path, quiet = quiet)
 6.           └─pkgload:::deps_check_installed(path, depends)
 7.             └─rlang::check_installed(pkg, action = action, call = call)
 8.               └─rlang:::pkg_version_info(pkg, version = version, compare = compare)
 9.                 └─rlang::abort(msg, call = call)
```

I did a bit of debugging and found out that the line which throws this error is present in the `pkg_version_info` function

```
if (!all(detect_na(info$cmp) | info$cmp %in% c(">", ">=", "<", "<="))) {
    msg <- sprintf(
      '%s must be one of ">", ">=", "<", or "<=".',
      format_arg("compare")
    )
    abort(msg, call = call)
  }
```
https://github.com/r-lib/rlang/blob/main/R/session.R#L133

Appending `==` to `info$cmp %in% c(">", ">=", "<", "<=")` seems to do the trick. However, my question is whether this is a bug or it was done on purpose?

**Comment by lionel-** [comment]
This was done on purpose, but we could relax this. Can you explain the context about using `==` instead of `>=` in your version requirements?

**Comment by KryeKuzhinieri** [comment]
A couple of times we got conflicts and breaks in production due to a couple of packages. Since we were using `>=`, it was extremely hard to debug at what point and which versions the problematic packages were causing the issues. Due to this, we decided to stick with `==` for simplicity reasons after testing the package versions. So, once in a couple of months, we update all the packages to their newest versions on Sandbox and then, once confirmed that all is working fine, we deploy them to production.
